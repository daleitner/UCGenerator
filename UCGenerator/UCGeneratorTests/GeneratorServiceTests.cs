using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using UCGenerator;
using UCGenerator.Services;
using UCGenerator.Statics;

namespace UCGeneratorTests
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class GeneratorServiceTests
	{
		private Mock<IDataService> dataService;

		[TestInitialize]
		public void Setup()
		{
			this.dataService = new Mock<IDataService>();
		}

		[TestMethod]
		public void VerifyGenerateMethod()
		{
			var service = new GeneratorService(this.dataService.Object);
			var projectFolder = "C:\\Projects\\TestProject";
			var nameSpace = "TestProject";
			var name = "Person";
			var controls = GenerateControls();
			var path = service.GenerateAbsolutePath(projectFolder, nameSpace);
			service.Generate(controls, projectFolder, nameSpace, name);
			this.dataService.Verify(x => x.CreateFile(service.GenerateUserControl(controls, nameSpace, name), 
				path + name + "UserControl.xaml"), Times.Once, "Usercontrol was not created");
			this.dataService.Verify(x => x.CreateFile(service.GenerateBackgroundFile(nameSpace, name),
				path + name + "UserControl.xaml.cs"), Times.Once, "Background File was not created");
			this.dataService.Verify(x => x.CreateFile(service.GenerateViewModel(controls, nameSpace, name),
				path + name + "ViewModel.cs"), Times.Once, "ViewModel was not created");
			this.dataService.Verify(x => x.CreateFile(service.GenerateControllerInterface(nameSpace, name),
				path + "I" + name + "Controller.cs"), Times.Once, "Controller Interface was not created");
			this.dataService.Verify(x => x.CreateFile(service.GenerateController(nameSpace, name),
				path + name + "Controller.cs"), Times.Once, "Controller was not created");
		}

		[TestMethod]
		public void VerifyGenerateAbsolutePath()
		{
			var service = new GeneratorService(this.dataService.Object);
			var projectFolder = "C:\\Projects\\TestProject";
			var nameSpace = "TestProject";
			var path = service.GenerateAbsolutePath(projectFolder, nameSpace);
			Approvals.Verify("Path= " + path);
		}

		[TestMethod]
		public void VerifyGenerateAbsolutePathWithSubNameSpace()
		{
			var service = new GeneratorService(this.dataService.Object);
			var projectFolder = "C:\\Projects\\TestProject\\";
			var nameSpace = "TestProject.TestAssembly";
			var path = service.GenerateAbsolutePath(projectFolder, nameSpace);
			Approvals.Verify("Path= " + path);
		}

		[TestMethod]
		public void VerifyGenerateUserControl()
		{
			var service = new GeneratorService(this.dataService.Object);
			var nameSpace = "TestProject";
			var name = "Person";
			var file = service.GenerateUserControl(GenerateControls(), nameSpace, name);
			Approvals.Verify(file);
		}

		[TestMethod]
		public void VerifyGenerateBackgroundFile()
		{
			var service = new GeneratorService(this.dataService.Object);
			var nameSpace = "TestProject";
			var name = "Person";
			var file = service.GenerateBackgroundFile(nameSpace, name);
			Approvals.Verify(file);
		}

		[TestMethod]
		public void VerifyGenerateViewModel()
		{
			var service = new GeneratorService(this.dataService.Object);
			var nameSpace = "TestProject";
			var name = "Person";
			var file = service.GenerateViewModel(GenerateControls(), nameSpace, name);
			Approvals.Verify(file);
		}

		[TestMethod]
		public void VerifyGenerateControllerInterface()
		{
			var service = new GeneratorService(this.dataService.Object);
			var nameSpace = "TestProject";
			var name = "Person";
			var file = service.GenerateControllerInterface(nameSpace, name);
			Approvals.Verify(file);
		}

		[TestMethod]
		public void VerifyGenerateController()
		{
			var service = new GeneratorService(this.dataService.Object);
			var nameSpace = "TestProject";
			var name = "Person";
			var file = service.GenerateController(nameSpace, name);
			Approvals.Verify(file);
		}

		private List<WPFControl> GenerateControls()
		{
			var viewModel = new MainViewModel(this.dataService.Object, new GeneratorService(this.dataService.Object));
			viewModel.AddCommand.Execute(null);
			viewModel.Controls.Last().PropertyName = "Name";
			viewModel.Controls.Last().Type = TypeEnum.TextBox;
			viewModel.Controls.Last().Bindings.ForEach(x => x.IsBound = true);
			viewModel.AddCommand.Execute(null);
			viewModel.Controls.Last().PropertyName = "Path";
			viewModel.Controls.Last().Type = TypeEnum.Label;
			viewModel.Controls.Last().Bindings.ForEach(x => x.IsBound = true);
			viewModel.AddCommand.Execute(null);
			viewModel.Controls.Last().PropertyName = "Players";
			viewModel.Controls.Last().Type = TypeEnum.ListBox;
			viewModel.Controls.Last().Bindings.ForEach(x => x.IsBound = true);
			viewModel.AddCommand.Execute(null);
			viewModel.Controls.Last().PropertyName = "Start";
			viewModel.Controls.Last().Type = TypeEnum.Button;
			viewModel.Controls.Last().Bindings.ForEach(x => x.IsBound = true);
			viewModel.AddCommand.Execute(null);
			viewModel.Controls.Last().PropertyName = "Colors";
			viewModel.Controls.Last().Type = TypeEnum.ComboBox;
			viewModel.Controls.Last().Bindings.ForEach(x => x.IsBound = true);
			viewModel.AddCommand.Execute(null);
			viewModel.Controls.Last().PropertyName = "Image";
			viewModel.Controls.Last().Type = TypeEnum.ContentPresenter;
			viewModel.Controls.Last().Bindings.ForEach(x => x.IsBound = true);
			viewModel.AddCommand.Execute(null);
			viewModel.Controls.Last().PropertyName = "Players";
			viewModel.Controls.Last().Type = TypeEnum.ItemsControl;
			viewModel.Controls.Last().Bindings.ForEach(x => x.IsBound = true);
			return viewModel.Controls.ToList();
		}
	}
}
