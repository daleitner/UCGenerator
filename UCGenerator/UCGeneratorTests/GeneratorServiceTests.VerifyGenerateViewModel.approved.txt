using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Input;
using Base;
using System.Windows;

namespace TestProject
{
	public class PersonViewModel : ViewModelBase
	{
		private string name;
		
		private bool isNameEnabled;
		private Visibility nameVisibility;
		private RelayCommand nameEnterCommand;
		private string path;
		private bool isPathEnabled;
		private Visibility pathVisibility;
		private List<object> players;
		private object selectedPlayer;
		private bool isPlayersEnabled;
		private Visibility playersVisibility;
		private RelayCommand playersEnterCommand;
		private RelayCommand playersEnterCommand;
		private string start;
		private RelayCommand startCommand;
		private Visibility startVisibility;
		private List<object> colors;
		private object selectedColor;
		private bool isColorsEnabled;
		private Visibility colorsVisibility;
		private object image;
		private Visibility imageVisibility;
		private List<object> players;
		private Visibility playersVisibility;
		public PersonViewModel()
		{
			this.players = new ObservableCollection<object>();
			this.colors = new ObservableCollection<object>();
			this.players = new ObservableCollection<object>();
		}

		public string Name
		{
			get
			{
				return this.name;
			}
			set
			{
				this.name = value;
				OnPropertyChanged(nameof(this.Name));
			}
		}

		public  Name
		{
			get
			{
				return this.name;
			}
			set
			{
				this.name = value;
				OnPropertyChanged(nameof(this.Name));
			}
		}

		public bool IsNameEnabled
		{
			get
			{
				return this.isNameEnabled;
			}
			set
			{
				this.isNameEnabled = value;
				OnPropertyChanged(nameof(this.IsNameEnabled));
			}
		}

		public Visibility NameVisibility
		{
			get
			{
				return this.nameVisibility;
			}
			set
			{
				this.nameVisibility = value;
				OnPropertyChanged(nameof(this.NameVisibility));
			}
		}

		public ICommand NameEnterCommand
		{
			get
			{
				return this.nameEnterCommand ?? (this.nameEnterCommand = new RelayCommand(param => Name()));
			}
		}

		public string Path
		{
			get
			{
				return this.path;
			}
			set
			{
				this.path = value;
				OnPropertyChanged(nameof(this.Path));
			}
		}

		public bool IsPathEnabled
		{
			get
			{
				return this.isPathEnabled;
			}
			set
			{
				this.isPathEnabled = value;
				OnPropertyChanged(nameof(this.IsPathEnabled));
			}
		}

		public Visibility PathVisibility
		{
			get
			{
				return this.pathVisibility;
			}
			set
			{
				this.pathVisibility = value;
				OnPropertyChanged(nameof(this.PathVisibility));
			}
		}

		public List<object> Players
		{
			get
			{
				return this.players;
			}
			set
			{
				this.players = value;
				OnPropertyChanged(nameof(this.Players));
			}
		}

		public object SelectedPlayer
		{
			get
			{
				return this.selectedPlayer;
			}
			set
			{
				this.selectedPlayer = value;
				OnPropertyChanged(nameof(this.SelectedPlayer));
			}
		}

		public bool IsPlayersEnabled
		{
			get
			{
				return this.isPlayersEnabled;
			}
			set
			{
				this.isPlayersEnabled = value;
				OnPropertyChanged(nameof(this.IsPlayersEnabled));
			}
		}

		public Visibility PlayersVisibility
		{
			get
			{
				return this.playersVisibility;
			}
			set
			{
				this.playersVisibility = value;
				OnPropertyChanged(nameof(this.PlayersVisibility));
			}
		}

		public ICommand PlayersEnterCommand
		{
			get
			{
				return this.playersEnterCommand ?? (this.playersEnterCommand = new RelayCommand(param => Players()));
			}
		}

		public ICommand PlayersEnterCommand
		{
			get
			{
				return this.playersEnterCommand ?? (this.playersEnterCommand = new RelayCommand(param => Players()));
			}
		}

		public string Start
		{
			get
			{
				return this.start;
			}
			set
			{
				this.start = value;
				OnPropertyChanged(nameof(this.Start));
			}
		}

		public ICommand StartCommand
		{
			get
			{
				return this.startCommand ?? (this.startCommand = new RelayCommand(param => Start()));
			}
		}

		public Visibility StartVisibility
		{
			get
			{
				return this.startVisibility;
			}
			set
			{
				this.startVisibility = value;
				OnPropertyChanged(nameof(this.StartVisibility));
			}
		}

		public List<object> Colors
		{
			get
			{
				return this.colors;
			}
			set
			{
				this.colors = value;
				OnPropertyChanged(nameof(this.Colors));
			}
		}

		public object SelectedColor
		{
			get
			{
				return this.selectedColor;
			}
			set
			{
				this.selectedColor = value;
				OnPropertyChanged(nameof(this.SelectedColor));
			}
		}

		public bool IsColorsEnabled
		{
			get
			{
				return this.isColorsEnabled;
			}
			set
			{
				this.isColorsEnabled = value;
				OnPropertyChanged(nameof(this.IsColorsEnabled));
			}
		}

		public Visibility ColorsVisibility
		{
			get
			{
				return this.colorsVisibility;
			}
			set
			{
				this.colorsVisibility = value;
				OnPropertyChanged(nameof(this.ColorsVisibility));
			}
		}

		public object Image
		{
			get
			{
				return this.image;
			}
			set
			{
				this.image = value;
				OnPropertyChanged(nameof(this.Image));
			}
		}

		public Visibility ImageVisibility
		{
			get
			{
				return this.imageVisibility;
			}
			set
			{
				this.imageVisibility = value;
				OnPropertyChanged(nameof(this.ImageVisibility));
			}
		}

		public List<object> Players
		{
			get
			{
				return this.players;
			}
			set
			{
				this.players = value;
				OnPropertyChanged(nameof(this.Players));
			}
		}

		public Visibility PlayersVisibility
		{
			get
			{
				return this.playersVisibility;
			}
			set
			{
				this.playersVisibility = value;
				OnPropertyChanged(nameof(this.PlayersVisibility));
			}
		}

		private void NameEnter()
		{
		}

		private void PlayersEnter()
		{
		}

		private void PlayersEnter()
		{
		}

		private void Start()
		{
		}

	}
}